### DISCUSSION TITLE
vercel build cd folder and run command not working

### DISCUSSION THREAD

**Discussion by surjithctly** [vercel/vercel#6475] [Help] [answered]
I'm trying to deploy a next.js project. before that, I'm deploying sanity studio using this command. 

```
  "scripts": {
    "build": "npm run build:sanity && next build",
    "build:sanity": "echo 'Building Sanity to public/studio' && cd studio && npx sanity build ../public/studio -y && cd .."
  },
```

But its throwing error: 

```
/bin/sh: sanity: command not found
Command failed with exit code 127.
```

I have also tried with `npx @sanity/cli build`, `yarn build`, `yarn && sanity build` and many more combinations. But none of them worked. However, the same command works locally. 

So the issue is only in vercel platform. Any idea how to fix this?

**Comment by barelyhuman** [comment, marked as answer]
can I see the `dependencies` and `devDependencies` and `scripts` of the same?

**Reply by surjithctly** [reply]
I have `@sanity/cli` as dependencies in the root folder (ie next.js) and I have also the same dependencies in `/studio` folder. Script for studio folder below. 

```json
  "scripts": {
    "start": "sanity start",
    "build": "sanity build"
  },
```

My guess is vercel is not actually navigating to the folder `cd studio` instead its trying to run the command from same folder.

**Reply by surjithctly** [reply]
Because I'm getting this error when @sanity/cli is installed in root. 

```
18:58:03.824 | Building Sanity to public/studio
18:58:04.018 | @sanity/core not installed in current project
18:58:04.018 | Project-specific commands not available until you run `sanity install`
18:58:04.023 | Error: Command "build" is not available outside of a Sanity project context.
```

**Reply by barelyhuman** [reply]
Cool, I'm going to create a similar structure and check this

**Reply by surjithctly** [reply]
Thanks, man! appreciate your help.

**Reply by barelyhuman** [reply]
So I didn't do much,  but it worked for me as intented. 
Changes I made was. 

on your root folder. 
```
{
  "prebuild":"cd studio;yarn;yarn build",
  "build":"next build"
}
```

Could you test that out and let me know if it worked for you?

**Reply by surjithctly** [reply]
It is throwing `Module not found` error now.

**Reply by barelyhuman** [reply]
For which package? the cli?

**Reply by surjithctly** [reply]
cool, it worked now, it was nother npm package. Was added as `devdependancy` now moved to normal dependency and it worked. 

Thanks a lot man. 

btw, do you know why prebuild worked but not "&&" as I added previously?

**Reply by barelyhuman** [reply]
just a hunch but I'm guessing the vercel platform defaults to `next build` on a next project and that is all that was running, if you want your upper scripts to work you can override the setting in the project settings and force it to run `npm run build`, that should also work. 

As for why the prebuild ran, it ran as it's an npm hook that caused the execution. 

Just a hunch, I could be totally wrong since I haven't gotten to the point of reading through their life cycle implementation based on each template yet.

**Reply by surjithctly** [reply]
Thank you. I'm glad it worked :)
